Lecture2/Task5/Task5L2.java

```diff
@@ -0,0 +1,154 @@
+            if (Arrays.equals ( nameArray , starNameArray ))
+                printWholeWord ( starNameArray , starLastnameArray );
+            else printRightMove ( starNameArray , starLastnameArray );
+        else if (checkRightMoveLastname)
+            if (Arrays.equals ( lastnameArray , starLastnameArray ))
+                printWholeWord ( starNameArray , starLastnameArray );
+            else printRightMove ( starNameArray , starLastnameArray );
+        else printWrongMove ( );
+    }
+
+    private static void finishGameValidation ( boolean checkResultName , boolean checkResultLastname ) {
+        if (checkResultName && checkResultLastname)
+            System.out.print ( "\nWonderful!!! You did it!!! See you later...\n" );
+    }
+
+    private static boolean letterMoveValidation ( String choice , int arraylength , char[] array , char[] starArray , boolean checkRightMove ) {
+        for ( int i = 0 ; i < arraylength ; i++ )
+            if (choice.equalsIgnoreCase ( Character.toString ( array[ i ] ) )) {
+                starArray[ i ] = array[ i ];
+                checkRightMove = true;
+            }
+        return checkRightMove;
+    }
+
+    private static void letterProcessing ( char[] nameArray , char[] starNameArray , String choice , char[] lastnameArray , char[] starLastnameArray , List <String> listOfLetters ) {
+        boolean checkRightMoveName = false;
+        boolean checkRightMoveLastname = false;
+        if (!Arrays.equals ( nameArray , starNameArray ))
+            checkRightMoveName = letterMoveValidation ( choice , nameArray.length , nameArray , starNameArray , checkRightMoveName );
+        if (!Arrays.equals ( lastnameArray , starLastnameArray ))
+            checkRightMoveLastname = letterMoveValidation ( choice , lastnameArray.length , lastnameArray , starLastnameArray , checkRightMoveLastname );
+        listOfLetters.add ( choice );
+        letterMessageValidation ( checkRightMoveName , checkRightMoveLastname , nameArray , lastnameArray , starNameArray , starLastnameArray );
+    }
+
+    private static void wordMessageValidation ( boolean checkRightMove , char[] starNameArray , char[] starLastnameArray ) {
+        if (checkRightMove)
+            printWholeWord ( starNameArray , starLastnameArray );
+        else printWrongMove ( );
+    }
+
+    private static boolean wordMoveValidation ( char[] array , char[] starArray ) {
+        System.arraycopy ( array , 0 , starArray , 0 , array.length );
+        return true;
+    }
+
+    private static void wordProcessing ( char[] nameArray , char[] starNameArray , String choice , String name , char[] lastnameArray , char[] starLastnameArray , String lastname , List <String> listOfWords ) {
+        boolean checkRightMove = false;
+        if (!Arrays.equals ( nameArray , starNameArray ) && choice.length ( ) == name.length ( ) && choice.equalsIgnoreCase ( name ))
+            checkRightMove = wordMoveValidation ( nameArray , starNameArray );
```

Comment:
not good name for variable, checkRightMove. isWordOpened -> more or less reasonable.

Reply:
fixed